### PR TITLE
support static shape in concatenate and filter in TGI

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?
add static shape support in concatenate and filter

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: text-generation-inference/server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -202,37 +202,51 @@ def filter(self, request_ids: List[int]) -> Optional["CausalLMBatch"]:
...
+            self.attention_mask = self.attention_mask[
+                keep_indices,
+                -(self.padding_right_offset + max_input_length) : (
+                    self.attention_mask.shape[1] - self.padding_right_offset
+                )
+                + new_padding_right_offset,
+            ]
 
         # Ensure that past_key_values tensors can be updated in-place
+        kv_tuple = False
```
Comment (regisss) [inline text-generation-inference/server/text_generation_server/models/causal_lm.py:217]: Why is this necessary now? Are layers automatically converted into lists at some point?

File: text-generation-inference/server/text_generation_server/models/causal_lm.py [hunk 2]
```diff
@@ -348,10 +362,12 @@ def concatenate(cls, batches: List["CausalLMBatch"]) -> "CausalLMBatch":
             # BLOOM Keys:   [batch_size * num_heads, head_dim, seq_length]
             # BLOOM Values: [batch_size * num_heads, seq_length, head_dim]
             # And ensure that we can update tensors in-place
+            kv_tuple = False
```
Comment (regisss) [inline text-generation-inference/server/text_generation_server/models/causal_lm.py:365]: Same

Comment (sywangyi) [reply text-generation-inference/server/text_generation_server/models/causal_lm.py:365]: in https://github.com/huggingface/optimum-habana/pull/389/files#diff-244871853752a7a019853c852803a797a9a14dddb3d9d859ecbae4b316ed6974R457

File: text-generation-inference/server/text_generation_server/models/causal_lm.py (hunk 1 above)
Comment (sywangyi) [reply text-generation-inference/server/text_generation_server/models/causal_lm.py:217]: in https://github.com/huggingface/optimum-habana/pull/389/files#diff-244871853752a7a019853c852803a797a9a14dddb3d9d859ecbae4b316ed6974R247. the past_key_value type should be exactly same with the case in single batch. or else hpu graph compile will meet type check error.

**Review by regisss — APPROVED** [review]
LGTM
